From 568037ae874d787e3b8ccc8a9392051362be5ea9 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@gmail.com>
Date: Wed, 9 Jan 2019 12:03:18 +1100
Subject: [PATCH 1/3] Remove source fallback for 3rd-party builds

eopkg is only used for 3rd party. 3rd party is for non-distributable files.
Therefore a fallback will never exist and only creates a confusing error
message for users that we no longer host the file when the original location
is removed.
---
 pisi/sourcearchive.py | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/pisi/sourcearchive.py b/pisi/sourcearchive.py
index 400efb8..c8325e7 100644
--- a/pisi/sourcearchive.py
+++ b/pisi/sourcearchive.py
@@ -65,20 +65,11 @@ class SourceArchive:
                 else:
                     pisi.fetcher.fetch_url(self.url, ctx.config.archives_dir(), self.progress)
             except pisi.fetcher.FetchError:
-                if ctx.config.values.build.fallback:
-                    self.fetch_from_fallback()
-                else:
-                    raise
+                raise
 
             ctx.ui.info(_("Source archive is stored: %s/%s")
                 % (ctx.config.archives_dir(), self.url.filename()))
 
-    def fetch_from_fallback(self):
-        archive = os.path.basename(self.url.get_uri())
-        src = os.path.join(ctx.config.values.build.fallback, archive)
-        ctx.ui.warning(_('Trying fallback address: %s') % src)
-        pisi.fetcher.fetch_url(src, ctx.config.archives_dir(), self.progress)
-
     def fetch_from_mirror(self):
         uri = self.url.get_uri()
         sep = uri[len("mirrors://"):].split("/")
-- 
2.20.1

